doc/logging/persistent_logger/persistent_loglevel.md

```diff
@@ -0,0 +1,217 @@
+- On startup: we will load the loglevel_db.json file into the LOGLEVEL DB.
+
+
+## 3.2 Persistent logger flow
+
+- Each component has a singleton Logger object with a log level property and listener thread.
+- On startup:
+  - The loglevel_db.json file is loaded into the LOGLEVEL DB during the init of database docker.
+  - The loglevel property is set accordingly to the loglevel value on the LOGLEVEL DB.
+- When a component writes a log message, the Logger writes the message only if the loglevel of the message is above the current loglevel property.
+- When the user wants to set a new log level to a component, he uses the "swssloglevel" CLI command. The "swssloglevel" script sets the new verbosity to the LOGLEVEL DB (database #3).
+- The LOGLEVEL DB change triggers an event, which is caught by the listener thread.
+- The listener thread changes the loglevel property of the Logger.
+- The user can use the "log-level save" command to save the current loglevel and make it persistent. It will copy the LOGLEVEL DB content into the loglevel_db.json.
+In addition to the log level, the LOGLEVEL DB contains the log output file. After "log-level save" the log output will be persistent to reboot too.
+
+### 3.2.1 Return to default log level
+
+  To return to the default log level, the user will remove the loglevel_db.json and reboot the switch.
+
+
+
+
+![persistent logger flow](/doc/logging/persistent_logger/persistent_logger.png)
+
+
+
+
+
+# 4 Flows
+
+## 4.1 Logger init flow
+
+When the system startup and the Database container initialize, we will load the loglevel_db.json into the LOGLEVEL DB (similar to the config_db.json). If the loglevel.json file is deleted, the system will generate a new loglevel_db.json file with default log level values.
+
+  - There is a concern about the log level of messages written before the database container is initialized (if it exists). From my understanding, the RSYSLOG-CONFIG is up only after the DATABASE container, so there won't be logs before the loglevel_db.json loads into the LOGLEVEL DB. In case there are messages before the loading, the messages will be in the default log level.
+
+
+## 4.2 "config reload" and "config save" flows
+
+  When the user runs "config reload" or "config save", it will not affect the loglevel state.
+   - If the user wants to save its log level configuration, he will use the dedicated CLI command "log-level save".
+   - We will **not** create similar CLI command to "config reload". The "config reload" command is necessary for features that can only configure from the config_db.json; this ability is unnecessary in the persistent loglevel feature.
+
+
+# 5 Warm Boot Support
+  
+  In today's implementation, we don't flush the LOGLEVEL DB before warm-boot, which means that if the user configures some loglevel (for example, debug), after warm-boot, the system startup with the same configurable loglevel (debug).
```

Comment:
In today's implementation --> with current implementation 
warm-boot --> warm-reboot

Reply:
Done